File: Games/BombermanGame.Shared/Interfaces/IPlayerInfo.cs
namespace Games.BombermanGame.Shared.Interfaces
{
    public interface IPlayerInfo : IPositionItem
    {
        int Score { get; }
        string Nickname { get; }
        Action? OnScoreUpdated { set; }

        void BlowUpPlayer();
        void PlayerBlowUpDestroyableCell();
    }
}

File: BombermanClient_C_PlusPlus/Games/BombermanGame.Shared/Interfaces/IPlayerInfo.cpp
//
// Created by Andrey on 02.12.2023.
//

#include "IPlayerInfo.h"

namespace Games_BombermanGame_Shared_Interfaces {
    int IPlayerInfo::GetScore() const {
        return _score;
    }

    const string IPlayerInfo::GetNickname() const {
        return _nickname;
    }

    void IPlayerInfo::SetOnScoreUpdated(void (*onScoreUpdated)()) {
        _onScoreUpdated = onScoreUpdated;
    }

    IPlayerInfo::IPlayerInfo(int x, int y, int score, string nickname) :
            IPositionItem(x, y),
            _score(score),
            _nickname(nickname) {}
} // Games_BombermanGame_Shared_Interfaces